### PR TITLE
ARROW-1147: [C++] Allow optional vendoring of flatbuffers in plasma

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: cpp/src/plasma/CMakeLists.txt [hunk 1]
```diff
@@ -59,8 +63,6 @@ add_custom_command(
 
 add_custom_target(gen_plasma_fbs DEPENDS ${PLASMA_FBS_OUTPUT_FILES})
 
-add_dependencies(gen_plasma_fbs flatbuffers_ep)
```
Comment (wesm) [inline cpp/src/plasma/CMakeLists.txt]: Does putting the dependency before the creation of the target cause any problems?

Comment (cpcloud) [inline cpp/src/plasma/CMakeLists.txt]: Good point, I'll move things around a bit.

**Comment by wesm** [comment]
Appveyor build: https://ci.appveyor.com/project/cpcloud/arrow/build/1.0.186

**Comment by wesm** [comment]
+1

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**